storage/innobase/page/page0page.cc

```diff
@@ -806,16 +805,17 @@ page_copy_rec_list_start(
 			/* Before trying to reorganize the page,
 			store the number of preceding records on the page. */
 			ulint ret_pos = page_rec_get_n_recs_before(ret);
+
 			/* Before copying, "ret" was the predecessor
 			of the predefined supremum record.  If it was
 			the predefined infimum record, then it would
 			still be the infimum, and we would have
 			ret_pos == 0. */
-			if (UNIV_UNLIKELY(!ret_pos
-					  || ret_pos == ULINT_UNDEFINED)) {
+			if (UNIV_UNLIKELY(ret_pos == ULINT_UNDEFINED)) {
```

Comment:
The comment here looks OK; no need to change it while changing the code. Removing the incorrect condition seems to be the actual bug fix.